common/Message.hpp

```diff
@@ -199,7 +199,9 @@ class Message
     std::string getForwardToken(const char* buffer, int length)
     {
         std::string forward = COOLProtocol::getFirstToken(buffer, length);
-        return (forward.find('-') != std::string::npos ? forward : std::string());
+        if (forward.find('-') != std::string::npos)
```

Comment:
Nit: while I like the new version as it's more obvious to step through it in gdb, is this here to work around some Coverity bug or I miss something and indeed the new version does less copying?

Reply:
I don't think its a coverity bug, in the sense that coverity basically recommends in its warning to have:

return (forward.find('-') != std::string::npos ? std::move(forward) : std::string());

which fits with e.g. https://fekir.info/post/guaranteed-copy-elision/ "the ternary operator forces a copy in a return statement (return expr ? a : b), unless both a and b are temporaries"

But adding explicit std::move to returns feels like something to be avoided: https://pvs-studio.com/en/blog/terms/6516/

So I'm drawn to "An if statement seems to be a superior alternative."

Reply:
I see now, thanks.